### ISSUE TITLE
DocComment sniff more fixing capabilities

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#392]
Here is the DocBlock:

``` php
/**
 * Checks user permission to execute given $event
 *
 * @param  kEvent $event
 * @return bool
 * @access public
 */
```

What immediately can be fixed in it:
1. empty line between `@param` and `@return` because `param` tag group needs to be separated
2. replace `bool` to `boolean` (maybe it's another sniff though)

**Comment by gsherwood** [comment]
The @return type one is easy, so I've committed that.

The other change would be done by the DocComment sniff and it would need to rewrite the entire comment to make it work because it is just checks if tags are grouped together. If it got something like this:
```
/**
 * @return ...
 * @param ...
 * @access ...
 * @param ...
 * @since ...
 */
```
It couldn't just add newlines, or else you would get this:
```
/**
 * @return ...
 *
 * @param ...
 *
 * @access ...
 *
 * @param ...
 *
 * @since ...
 */
```
You actually need it to rewrite the entire comment like this:
```
/**
 * @return ...
 *
 * @param ...
 * @param ...
 *
 * @access ..
 * @since ...
 */
```
Which is a lot harder than a simple replacement.

**Comment by aik099** [comment]
So the more explicit stuff needs be made configurable in the sniff:
* what groups are there
* what tags allowed in what groups (can there be several tag types in same group?)
* what is the order of the groups
* what is the tag order in same group

Then we can safely rearrange what's needed.

**Comment by gsherwood** [comment]
Closing this as the change is quite significant and risky + there are no PRs or additional support for this idea.
